text/3537-msrv-resolver.md

```diff
@@ -0,0 +1,872 @@
+
+```toml
+[build]
+resolver.precedence = "rust-version"  # Default with `v3`
+```
+with potential values being:
+- `maximum`: behavior today (default for v1 and v2 resolvers)
+  - Needed for [verifying latest dependencies](https://doc.rust-lang.org/nightly/cargo/guide/continuous-integration.html#verifying-latest-dependencies)
+- `minimum` (unstable): `-Zminimal-versions`
+  - As this just just precedence, `-Zdirect-minimal-versions` doesn't fit into this
+- `rust-version`:  what is defined in the package (default for v3 resolver)
+- `rust-version=` (future possibility)
+  - `package`: long form of `rust-version`
+  - `rustc`: the current running version
+    - Needed for "separate development / publish MSRV" workflow
+  - `<x>[.<y>[.<z>]]`: manually override the version used
+
+If a `rust-version` value is used, we'd switch to `maximum` when `--ignore-rust-version` is set.
+This will let users effectively pass `--ignore-rust-version` to all commands,
+without having to support the flag on every single command.
+
+## `cargo build`
+
+The MSRV-compatibility build check will be demoted from an error to a `deny`-by-default workspace
+[diagnostic](https://github.com/rust-lang/cargo/issues/12235),
+allowing users to intentionally use dependencies on an unsupported (or less supported) version of Rust
+without requiring `--ignore-rust-version` on every invocation.
+
+## `cargo update`
+
+`cargo update` will inform users when an MSRV or semver incompatible version is available.
+`cargo update -n` will also report this information so that users can check on the status of this at any time.
+
+**Note:** other operations that cause `Cargo.lock` entries to be changed (like
+editing `Cargo.toml` and running `cargo check`) may not inform the user.
+If they want to check the status of things, they can run `cargo update -n`.
+
+Users may pass
+- `--ignore-rust-version` to pick the latest dependencies, ignoring all `rust-version` fields (your own and from dependencies)
+- `--update-rust-version` to pick the `rustc --version`-compatible dependencies, updating your `package.rust-version` if needed to match the highest of your dependencies
+
+## `cargo add`
+
+`cargo add <pkg>` (no version) will pick a version requirement that is low
+enough so that when it resolves, it can pick a dependency that is
+MSRV-compatible.
+`cargo add` will warn when it does this.
+
+Users may pass
+- `--ignore-rust-version` to pick the latest dependencies, ignoring all `rust-version` fields (your own and from dependencies)
+- `--update-rust-version` to pick the `rustc --version`-compatible dependencies, updating your `package.rust-version` if needed to match the highest of your dependencies
+
+## `cargo publish`
+
+We will add `"auto"` for `package.rust-version`.
+On publish, `rustc --version` will replace `"auto"`.
+
+`cargo new` will include `package.rust-version = "auto"`.
```

Comment:
for clarification: I assume the published `Cargo.toml` will contain a proper version, not `auto`, correct? What about `Cargo.toml.orig`? I assume that will contain `rust-version = "auto"?

Reply:
Yes, the published `Cargo.toml` will not have `auto` and `Cargo.toml.orig` will.

As for the name, I went with it mostly because we use it elsewhere.  I take it as "cargo will worry about it, you don't have to".  Based on our definition of "support", guessing an MSRV is out of scope.

Otherwise, I have no strong opinion and will leave this to the cargo team to decide.